### PR TITLE
[Fix #101, #103] LiteralAsActualArgument parsing and autocorrect errors

### LINKED ISSUE #101: Minitest/LiteralAsActualArgument cop error
> An error occurred while Minitest/LiteralAsActualArgument cop was inspecting XXXXXXXX:65.

On that line is:

```ruby
assert_equal releases_count, form.all_releases_dataset.all.size
```

Other lines that cause error this include:

```ruby
assert_equal(*channels_normal_attrs)
assert_equal current_path, "/releases/distribute"
# and several more, but all of them in my case either splat args, or omit parens.
```

--------

## Expected behavior

Should not error when parsing.

## Actual behavior

Errors when parsing.

## Steps to reproduce the problem

I'm guessing it could be repro'd in a test here https://github.com/rubocop-hq/rubocop-minitest/blob/master/test/rubocop/cop/minitest/literal_as_actual_argument_test.rb

I will try to repro in a test next if I have time.

## RuboCop version

```
bash-5.0# bundle exec rubocop -V
0.87.1 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.5.3 x86_64-linux-musl)
```

### LINKED ISSUE #103: Bug: Minitest/LiteralAsActualArgument Autocorrected should add Hash literal curly braces
When moving 2nd arg to be 1st arg, the curly braces that were optional because the Hash was the last argument are no longer optional, and the result is invalid Ruby syntax.

--------

## Expected behavior

This 

```ruby
assert_equal(actual, :expected => "stuff")
```

should be autocorrected to

```ruby
assert_equal({ :expected => "stuff" }, actual)
```

## Actual behavior

This 

```ruby
assert_equal(actual, :expected => "stuff")
```

becomes autocorrected (incorrectly) to:

```ruby
assert_equal(:expected => "stuff", actual)
```

## Steps to reproduce the problem

Provide a Hash without curly braces as the 2nd arg to `assert_equal` and run `Minitest/LiteralAsActualArgument` autocorrect.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
bash-5.0# bundle exec rubocop -V
0.87.1 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.5.3 x86_64-linux-musl)
```

### PR DESCRIPTION
**Pull request by TSMMark**
Fixes #101 and fixes #103. More info in those issues

-----------------

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,10 @@
 * [#84](https://github.com/rubocop-hq/rubocop-minitest/pull/84): New cops `AssertKindOf` and `RefuteKindOf` check for use of `assert_kind_of`/`refute_kind_of` instead of `assert(foo.kind_of?(Class))`/`refute(foo.kind_of?(Class))`. ([@fatkodima][])
 * [#85](https://github.com/rubocop-hq/rubocop-minitest/pull/85): Add autocorrect to `Rails/AssertEmptyLiteral` cop. ([@fatkodima][])
 
+### Bug fixes
+
+* [#101](https://github.com/rubocop-hq/rubocop-minitest/issues/101): Fix parsing error for `Minitest/LiteralAsActualArgument` when using splat or when invoking without parentheses. ([@tsmmark][])
```
Comment (koic) [inline CHANGELOG.md:20]: Good catch! On the other hand, `Minitest/LiteralAsActualArgument` cop is an unreleased feature, so the changelog entry is not required.

Comment (TSMMark) [reply CHANGELOG.md]: Right, of course. Updated

**Comment by TSMMark** [comment]
@koic thanks for the quick review! I've made the requested update and rebased to squash the commits together. I figured I might as well fix #103 here as well so I included that also.

File: lib/rubocop/cop/minitest/literal_as_actual_argument.rb [hunk 2]
```diff
@@ -18,30 +18,34 @@ module Minitest
...
+          add_offense(node, location: all_arguments_range(node)) if actual.recursive_basic_literal?
         end
 
         def autocorrect(node)
           expected, actual, message = *node.arguments
-          arguments = [actual.source, expected.source, message&.source].compact.join(', ')
 
           lambda do |corrector|
+            new_actual_source =
+              if actual.is_a?(RuboCop::AST::HashNode) && actual.source[0] != '{'
```
Comment (koic) [inline lib/rubocop/cop/minitest/literal_as_actual_argument.rb:39]: Can you use `hash_type?` and `braces?` methods?

```suggestion
              if actual.hash_type? && !actual.braces?
```

Comment (TSMMark) [reply lib/rubocop/cop/minitest/literal_as_actual_argument.rb:39]: Yes! I figured what I had was not idiomatic but I had some trouble finding good docs / examples for parser. Do you have a recommendation?

Will update and squash this in a few minutes.

Comment (TSMMark) [reply lib/rubocop/cop/minitest/literal_as_actual_argument.rb]: Done

Comment (koic) [reply lib/rubocop/cop/minitest/literal_as_actual_argument.rb]: Thank you!

>  I figured what I had was not idiomatic but I had some trouble finding good docs / examples for parser

For example, there are the following docs.

- https://docs.rubocop.org/rubocop-ast/index.html
- https://www.rubydoc.info/gems/rubocop-ast/RuboCop/AST
- https://www.rubydoc.info/github/whitequark/parser/Parser

And IMHO, code reading into RuboCop AST's may be useful.

**Pull request merged**